### ISSUE TITLE
Can't set new Transform to GifDrawable ( or change CornerRadius more  than once )

### ISSUE THREAD

**Issue by Bendor** [koral--/android-gif-drawable#450]
Hi!
I have a problem with GifDrawable _Transform_  :(
All current _Transform_  implementations have to receive and **hold** GifDrawable  bunds rect by API method _pl.droidsonroids.gif.transforms.Transform#onBoundsChange_
But GifDrawable  call this method olny in own method _onBoundsChange_, see:
```
@Override
	protected void onBoundsChange(Rect bounds) {
		mDstRect.set(bounds);
		if (mTransform != null) {
			mTransform.onBoundsChange(bounds);
		}
	}
```
So if I set new Transform to GifDrawable  at the moment when the GifDrawable  has already received bounds then my new Transform will never get bounds Rect, see:
```

	public void setTransform(@Nullable Transform transform) {
		mTransform = transform;
	}

OR

	public void setCornerRadius(@FloatRange(from = 0) final float cornerRadius) {
		mTransform = new CornerRadiusTransform(cornerRadius);
	}

```
GifDrawable   field  _Rect mDstRect_ is private and there no getter method, so I can't receive it to put in my new Transform.

In this case GifDrawable will draw nothing because Transform.onDraw will be draw to empty rect (for example see _pl.droidsonroids.gif.transforms.CornerRadiusTransform#onDraw_ 72 line).

This problem can be solved in several ways:
1. Change this methods in GifDrawable:

```
public void setTransform(@Nullable Transform transform) {
		mTransform = transform;
		if  ( mTransform != null)  {
		mTransform.onBoundsChange(mDstRect );
		}

	}
AND
	public void setCornerRadius(@FloatRange(from = 0) final float cornerRadius) {
		mTransform = new CornerRadiusTransform(cornerRadius);
		mTransform.onBoundsChange(mDstRect );
	}

```
2. Remove method _Transform.onBoundsChange_ and add param Rect dstRect to _Transform.onDraw_
In this case no needs hold Rect in every Transform implementation

Also it would be useful to add a method mDstRect getter method.

//-------------------------------------------------------------------//

And one more request: please change the access modifiers in pl.droidsonroids.gif.transforms.CornerRadiusTransform fields  from **private** to **protected**, it will be very useful for me!

Thanks!

**Comment by koral--** [comment]
> Remove method Transform.onBoundsChange and add param Rect dstRect to Transform.onDraw
In this case no needs hold Rect in every Transform implementation

Instead of removing I would add new methods and mark old ones deprecated, so it won't break backwards compatibility. I'll check it ASAP.

**Comment by koral--** [comment]
I've analyzed this issue deeper and decided to use option 1. - missing `onBoundsChange()` calls have been added. 
Making `onBoundsChange()` deprecated will cause that existing transforms which only need recompute something if bounds change need to be refactored.
I've also added `CornerRadiusTransform` accessors, except `mShader` which is implementation detail.

Changes available in version `1.2.9-SNAPSHOT` on maven snapshot repo.
